control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
 For such a RPC message, the first content frame MUST consist in a JSON-RPC compatible content, for example a single request object or a batch of request objects.
 
 
+### List of methods
+
+List of methods defined by LECO.
+You MAY implement the optional methods of this list and you MUST implement the methods obligatory for your type of Component.
+All methods implemented in a Component MUST adhere to this list or MUST have a name different to any method in this list.
+
+
+#### Component
+
+Any Component, i.e. any participant in the LECO protocol, MUST offer the following methods.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Component",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "pong",
+            "description": "Respond to a ping.",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        }
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
+Any Component MAY offer ANY of the following methods.
+Components SHOULD offer ``shut_down``.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Component-optional",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "set_log_level",
+            "params": [
+                {
+                    "name": "level",
+                    "schema": {
+                        "type": "integer"
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "shut_down",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
```

Comment:
Is this generally the way to express a method without return value?

Reply:
The method returns something, namely a `None` object.

If you do not specify a `result` object, the method will never send any response.
JSONRPC allows to send a message (called notification), which will not get a response, by not setting any `id`.


"The description of the result returned by the method. If defined, it MUST be a Content Descriptor or Reference Object. If undefined, the method MUST only be used as a [notification](https://www.jsonrpc.org/specification#notification)." (https://spec.open-rpc.org/#content-descriptor-object)

Reply:
> The method returns something, namely a None object.

Careful, a function returning `None` even when it has no `return` is a Pythonism. A C++ `void` function would return nothing. We should strive to have LECO be implementation-language-independent.

Is returning null used as some kind of confirmation in LECO? Are we using that returned object for anything (Sorry, I have forgotten this detail). If yes, should that be a success indicator or something? If no, should that method maybe be such a notification?

Reply:
Let's make it more hands on:

We have a server (in whatever language written).
If it offers a method which does not have a return value defined in the openrpc schema file, the server must not send a response after the method has been executed.
If the openrpc schema file defines `result: null`, then the software shall send a response with `"result": null` in ASCII code, if requested. Any language should be able to generate that ASCII code, whether the language has a `None` object or not.

So its all about the JSON definitions, not about the underlying language. I'm just applying these specs to our use case.